flaml/automl/model.py

```diff
@@ -1620,7 +1619,11 @@ def config2params(self, config: dict) -> dict:
         if max_depth == 0:
             params["grow_policy"] = params.get("grow_policy", "lossguide")
             params["tree_method"] = params.get("tree_method", "hist")
-        params["use_label_encoder"] = params.get("use_label_encoder", False)
+        # use_label_encoder is deprecated in 1.7.
+        if xgboost_version < "1.7.0":
+            params["use_label_encoder"] = params.get("use_label_encoder", False)
+        else:
+            assert "use_label_encoder" not in params, "use_label_encoder is deprecated in xgboost>=1.7.0"

```

Comment:
No need to assert here. Assert will slow down the training process.

Reply:
I was trying to find the bug. This assert doesn't fail, so "use_label_encoder" is indeed not in params with xgboost 2. 
So, why does error about "use_label_encoder" happen? Does any executor use xgboost < 1.7.0 for some reason?

Reply:
I see that the bug is fixed now. Then we can remove this assertion. I'll approve after that.

Reply:
> I was trying to find the bug. This assert doesn't fail, so "use_label_encoder" is indeed not in params with xgboost 2. So, why does error about "use_label_encoder" happen? Does any executor use xgboost < 1.7.0 for some reason?

In the macOS test, 1.7.6 is actually the version we want to use. But somehow the version in spark driver is upgraded to 2.0.0. The root cause is that in `test/automl/test_classification.py::fix test_sparse_matrix_xgboost`, it will first downgrade xgboost to 1.3.3 and then upgrade it to 2.0.0. Which makes in the spark test, the driver will have version 2.0.0 and executor will have 1.7.6 since the upgrade in `test_sparse_matrix_xgboost` only affects driver.